utils/build-toolchain

```diff
@@ -97,6 +97,7 @@ YEAR=$(date +"%Y")
 MONTH=$(date +"%m")
 DAY=$(date +"%d")
 TOOLCHAIN_VERSION="swift-LOCAL-${YEAR}-${MONTH}-${DAY}-a"
+DARWIN_TOOLCHAIN_VERSION="4.2.${YEAR}${MONTH}${DAY}"
```

Comment:
I'm concerned about this "4.2". The master branch isn't 4.2-ish anymore, but even without that it's one more place where a version number gets hardcoded.

Reply:
Difficult to know the alternative is other than adding another option.

Reply:
I don't think we want an alternative; we want it inferred from some other part of the compiler.

@aciidb0mb3r, anything you can share?

Reply:
It doesn't look like its using the version for anything very "important". It uses it to handle conflicts between alias names. It is fine as long as you use numbers separated by dots. Maybe just use `${YEAR}.${MONTH}.${DAY}`?

Reply:
Thanks, this seems to work fine (though it has the feeling of something that might break someone's assumptions further down the line.) New amend pushed.

Reply:
Should we do the same thing as the swift-syntax versions? `0.MAJOR_MINOR_PATCH.YEAR_MONTH_DAY`? Or maybe `0.0.YEAR_MONTH_DAY` for now, since we don't have a good way to get the version in.

Reply:
The version number is probably not going to matter in practice unless someone starts depending on it. I like Jordan's `0.0.YEAR_MONTH_DAY` idea though.

Reply:
0.0.YEAR_MONTH_DAY it is then, commit amended. New build system seems happy enough.
```
        <key>Version</key>
        <string>0.0.2018_09_25</string>
```